src/main/java/controllers/SwingBoard.java

```diff
@@ -157,88 +315,285 @@ private void displayEvent(Outcomes outcome) {
 
             case GAME_OVER:
                 endGame();
-                break;
+                return;
         }
 
     }
 
+
     public void promptPlayAgain() {
         if (JOptionPane.showConfirmDialog(null, "Do you want to play again?", "Play Again?",
                 JOptionPane.YES_NO_OPTION) == JOptionPane.YES_OPTION) {
-            resetGame();
+            return;
         } else {
             System.exit(0);
         }
     }
 
 
     public void resetGame() {
-
         frame.setVisible(false);
         frame.removeAll();
-        GameState newGame = state.resetGuesses();
+        GameState newGame = state.resetGame();
+        PlayerSelection playerSelection = new PlayerSelection();
         SwingBoard location = new SwingBoard(newGame, playerSelection);
         revalidate();
         repaint();
+        return;
+
+    }
+
+    public void getEndingDialogue() {
```

Comment:
I believe this can be private - check all access modifiers to try lock down code where possible